src/util/usePreLoader.ts

```diff
@@ -0,0 +1,45 @@
+import {Story} from "../interfaces";
+import {useEffect, useState} from "react";
+
+
+// Caches given Story[] using HTMLImageElement and HTMLVideoElement
+const cacheContent = async (contents: Story[]) => {
+	const promises = contents.map((content) => {
+		return new Promise(function (resolve, reject) {
+			if(!content.url) return
+
+			if(content.type === 'video') {
+				const video = document.createElement('video');
+				video.src = content.url;
+				video.onloadeddata = resolve;
+				video.onerror = reject;
+				return;
+			}
+
+			const img = new Image();
+			img.src = content.url;
+			img.onload = resolve;
+			img.onerror = reject;
+		})
+	})
+
+	await Promise.all(promises);
+}
+
+
+// Preloads images and videos from given Story[] using a cursor and preloadCount
+// Preload count is the number of images/videos to preload after the cursor
+// Cursor is the current index to start preloading from
+export const usePreLoader = (contents: Story[], cursor: number, preloadCount: number) => {
+	useEffect(() => {
+		const start = cursor + 1;
+		const end = cursor + preloadCount + 1;
+		const toPreload = contents.slice(start, end);
```

Comment:
I would suggest maintaining a set of URLs that have been preloaded, and then checking here and skipping if preloaded already.

Reply:
Yeah i did consider something like that... But I'm uncertain to how much performance it actually saves (compared to my laziness). Will see if i can find a smooth solution

Reply:
Okay, performance wasn't the primary aim though, right now it still makes a request if you check the network tab, but then browser cache kicks in and resolves it. It would be great if the packages does not make the request at all.